### ISSUE TITLE
Can not combine fields 

### ISSUE THREAD

**Issue by rhuss** [atlasmap/atlasmap-runtime#241]
```
09:40:51.949 [HttpClient@871550938-27] ERROR o.a.c.processor.DefaultErrorHandler - Failed delivery for (MessageId: ID-sfdb-4-3f61z-1508924344117-0-2 on ExchangeId: ID-sfdb-4-3f61z-1508924344117-0-1). Exhausted after delivery attempt: 1 caught: io.atlasmap.api.AtlasException: Combine requires positive Index value to be set on all inputFields inputField.path=/FirstName

Message History
---------------------------------------------------------------------------------------------------------------------------------------
RouteId              ProcessorId          Processor                                                                        Elapsed (ms)
[flow1             ] [flow1             ] [salesforce-on-create-component://syndesis_Lead_create?notifyForFields=ALL&noti] [      1176]
[flow1             ] [to1               ] [atlas:mapping-step-2.json                                                     ] [      1173]

Stacktrace
---------------------------------------------------------------------------------------------------------------------------------------
io.atlasmap.api.AtlasException: Combine requires positive Index value to be set on all inputFields inputField.path=/FirstName
	at org.apache.camel.component.atlasmap.AtlasEndpoint.onExchange(AtlasEndpoint.java:221)
	at org.apache.camel.impl.ProcessorEndpoint$1.process(ProcessorEndpoint.java:71)
	at org.apache.camel.util.AsyncProcessorConverterHelper$ProcessorToAsyncProcessorBridge.process(AsyncProcessorConverterHelper.java:61)
	at org.apache.camel.processor.SendProcessor.process(SendProcessor.java:148)
	at org.apache.camel.processor.RedeliveryErrorHandler.process(RedeliveryErrorHandler.java:548)
	at org.apache.camel.processor.CamelInternalProcessor.process(CamelInternalProcessor.java:201)
	at org.apache.camel.processor.Pipeline.process(Pipeline.java:138)
	at org.apache.camel.processor.Pipeline.process(Pipeline.java:101)
	at org.apache.camel.processor.CamelInternalProcessor.process(CamelInternalProcessor.java:201)
	at org.apache.camel.processor.Pipeline.process(Pipeline.java:138)
	at org.apache.camel.processor.Pipeline.access$100(Pipeline.java:43)
	at org.apache.camel.processor.Pipeline$1.done(Pipeline.java:157)
```

Maybe related to #240 ? This is however more urgent as it is described in the TP2 tutorial ..

**Comment by rhuss** [comment]
When I try to change from 1,2 to 0,1 I get the following error message:

![image](https://user-images.githubusercontent.com/99080/31992244-0d195b3e-b97a-11e7-9570-ec1fc6f2f538.png)

At least the error message is misleading.

**Comment by rhuss** [comment]
Also strange: The red cross on the right side jumps around randomly when dismissing the error message (but that's probably another issue in the UI)

![image](https://user-images.githubusercontent.com/99080/31992309-41d527b8-b97a-11e7-8fee-888ed1bc37fe.png)

This error also make no sense (its on the next line for the email which has no combine mapping): 

![image](https://user-images.githubusercontent.com/99080/31992330-54af5fb6-b97a-11e7-821d-401e7ab5265c.png)

**Comment by igarashitm** [comment]
* separate/combine index issue
  * This is supposed to be fixed by #242
* The red cross on the right side jumps around randomly when dismissing the error message 
  * I haven't seen but sounds like yet another UI issue
* combine mapping error shows up in another mapping detail
  * This is an existing bug, all the mapping error shows up in every mapping detail regardless of focused mapping - atlasmap/atlasmap-ui#112
